src/test/java/com/google/cloud/spanner/pgadapter/python/django/DjangoPgAggregates.java

```diff
@@ -0,0 +1,230 @@
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameter;
+import org.junit.runners.Parameterized.Parameters;
+
+@RunWith(Parameterized.class)
+@Category(PythonTest.class)
+public class DjangoPgAggregates extends DjangoTestSetup {
+
+  @Parameter public String host;
+
+  @Parameters(name = "host = {0}")
+  public static List<Object[]> data() {
+    return ImmutableList.of(new Object[] {"localhost"}, new Object[] {"/tmp"});
+  }
+
+  private static ResultSet createResultSet(List<String> rows, int numCols) {
+    ResultSet.Builder resultSetBuilder = ResultSet.newBuilder();
+
+    StructType.Builder builder = StructType.newBuilder();
+
+    for (int i = 0; i < 2 * numCols; i += 2) {
+      String colType = rows.get(i);
+      if (colType.equals("int")) {
+        builder.addFields(
+            Field.newBuilder()
+                .setType(Type.newBuilder().setCode(TypeCode.INT64).build())
+                .setName(rows.get(i + 1))
+                .build());
+
+      } else if (colType.equals("string")) {
+        builder.addFields(
+            Field.newBuilder()
+                .setType(Type.newBuilder().setCode(TypeCode.STRING).build())
+                .setName(rows.get(i + 1))
+                .build());
+      } else if (colType.equals("array")) {
+        builder.addFields(
+            Field.newBuilder()
+                .setType(
+                    Type.newBuilder()
+                        .setCode(TypeCode.ARRAY)
+                        .setArrayElementType(Type.newBuilder().setCode(TypeCode.STRING))
+                        .build())
+                .setName(rows.get(i + 1))
+                .build());
+      }
+    }
+    resultSetBuilder.setMetadata(
+        ResultSetMetadata.newBuilder().setRowType(builder.build()).build());
+    for (int i = 2 * numCols; i < rows.size(); i += numCols) {
+      ListValue.Builder rowBuilder = ListValue.newBuilder();
+      for (int j = i; j < i + numCols; ++j) {
+        if (rows.get(j).charAt(0) == '[') {
+          String[] array = rows.get(j).substring(1, rows.get(j).length() - 1).split(",");
+
+          ListValue.Builder listValueBuilder = ListValue.newBuilder();
+          for (String arrElement : array) {
+            listValueBuilder.addValues(Value.newBuilder().setStringValue(arrElement).build());
+          }
+          Value value = Value.newBuilder().setListValue(listValueBuilder.build()).build();
+          rowBuilder.addValues(value);
+        } else rowBuilder.addValues(Value.newBuilder().setStringValue(rows.get(j)).build());
+      }
+      resultSetBuilder.addRows(rowBuilder.build());
+    }
+    return resultSetBuilder.build();
+  }
+
+  @BeforeClass
+  public static void mockUnnecessarySql() {
```

Comment:
I think it would be better to add this to the base class instead of here, as this will probably be needed for other Python tests as well (and it cleans up this test, so it only contains code that is relevant for what we are actually testing here)

Reply:
done